ziti/default_collection.go

```diff
@@ -0,0 +1,80 @@
+/*
+	Copyright 2019 NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package ziti
+
+// Deprecated: DefaultCollection is deprecated and is included for legacy support.
```

Comment:
as part of this deprecation, maybe it makes sense to change
```
func NewDialerWithFallback(ctx context.Context, fallback Dialer) Dialer
```
and
```
func NewDialer() Dialer {
	return &dialer{}
}
```

and add the ziti context/CtxCollection into that private `dialer` that's returned. That way when `Dial(network, address string) (net.Conn, error)` is called, the ziti context is available to be used and `ForAllContexts` isn't necessary? That seems like it'd work nicely?

Reply:
Right now it's implicit, so when the consumer makes a new dialer, they aren't the one keeping/tracking the state. That'd be a downside, it is kinda convenient from the consuming side to not have to deal with it... dunno